### PR TITLE
Fix change-in-update warning from Tasks

### LINKED ISSUE #3566: [labs/task] Task improperly warns user about rendering
### Which package(s) are affected?

Task (@lit-labs/task)

### Description

Using the default `render` from `Task` results in a console warning.

image

### Reproduction

The following class warns user there is an issue, despite just following default examples.

```js
export class WorldPosts extends LitElement {
  posts: Post[] = [];

  private _apiTask = new Task(this, {
    task: async () => {
      console.log('demo time');
      return [
        {
          message: 'hello',
        },
      ];
    },
     // this will autorun, but not run automatically on any subsequent updates since it tracks no args
    args: () => [],
  });

  render() {
    return html`
      ${this._apiTask.render({
        complete: (posts: Post[]) => {
          const list = posts.map(post => html`<li>${post.message}</li>`);

          return html`<ul>
            ${list}
          </ul>`;
        },
        error: console.error,
      })}
    `;
  }
}
```

### Workaround

Possibly overwriting update methods? Not sure

### Is this a regression?

No or unsure. This never worked, or I haven't tried before.

### Affected versions

2.5.0

### Browser/OS/Node environment

Browser: Chrome
OS: 13.1 (22C65)

### PR DESCRIPTION
**Pull request by justinfagnani**
Fixes #3566 

The fix is to call the initial `host.requestUpdate()` (which is there to get the element to update and see the PENDING state of the task) in a microtask, so that it goes after the change-in-update check.

### CODE REVIEW

**Comment by e111077** [comment]
What if you need DOM access or willUpdate property resolution?

**Comment by justinfagnani** [comment]
@e111077 that's a great question. We don't have any tests that cover and guarantee that ability, so if we want to preserve it we should add one.

There may be another way to solve #3566 where we just don't call `host.requestUpdate()` synchronously. Let me check on that.

**Comment by justinfagnani** [comment]
@e111077 I fixed this by putting `host.requestUpdate()` in a microtask and added a test that Tasks can see side effects of `update()` to ensure that behavior.

File: .changeset/fast-apes-cross.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@lit-labs/task': minor
+---
+
+Tasks now run on hostUpdate() instead of hostUpdated()
```
Comment (e111077) [inline .changeset/fast-apes-cross.md:5]: Please update changelog, but looks good!

**Review by e111077 — CHANGES_REQUESTED** [review]
Wow great job!

**Review by e111077 — APPROVED** [review]

**Pull request merged**